### PR TITLE
[SPARK-9154][SQL] codegen StringFormat

### PR DESCRIPTION
**Pull request by tarekbecker**
Jira: https://issues.apache.org/jira/browse/SPARK-9154

fixes bug of #7546

@marmbrus I can't reopen the other PR, because I didn't closed it. Can you trigger Jenkins?

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringOperations.scala [hunk 1]
```diff
@@ -486,6 +486,10 @@ case class StringFormat(children: Expression*) extends Expression with CodegenFa
   private def format: Expression = children(0)
   private def args: Seq[Expression] = children.tail
 
+  override def inputTypes: Seq[AbstractDataType] =
+    StringType :: List.fill(children.size - 1)(AnyDataType)
```
Comment (tarekauel) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringOperations.scala]: I updated as well.

**Comment by marmbrus** [comment]
ok to test

**Comment by rxin** [comment]
It was my mistake -- I shouldn't have taken out the formatString with Columns, except the 1st argument should be a String.

**Comment by tarekauel** [comment]
Should I add it again?

**Comment by rxin** [comment]
Yes please.

**Comment by tarekauel** [comment]
@rxin Could you trigger Jenkins

**Comment by marmbrus** [comment]
ok to test

**Comment by rxin** [comment]
Jenkins, retest this please.

File: sql/core/src/main/scala/org/apache/spark/sql/functions.scala [hunk 2]
```diff
@@ -1743,6 +1743,17 @@ object functions {
 
   /**
    * Format strings in printf-style.
+   *
+   * @group string_funcs
+   * @since 1.5.0
+   */
+  @scala.annotation.varargs
+  def formatString(format: Column, arguments: Column*): Column = {
```
Comment (rxin) [inline sql/core/src/main/scala/org/apache/spark/sql/functions.scala]: i was thinking about a a version in which format is a string, but rest of the arguments are columns.

**Pull request closed without merging**